### PR TITLE
Voice public message format and time functionality

### PR DESCRIPTION
**Pull request by NicolaasDieBaas**

### CODE REVIEW

**Comment by NicolaasDieBaas** [comment]
@gsvr plz review

File: src/voice_public.js [hunk 1]
```diff
@@ -254,28 +254,35 @@ go.app = function() {
...
-                    if (messageType == "text") {
-                        return self.states.create('state_sms_menu');
+                .get_subscription_msg_type(self.im, self.im.user.answers.mother_id)
+                .then(function(msg_format) {
+                    self.im.user.set_answer('msg_format', msg_format);
+                    if (msg_format == "text") {
+                        return self.states.create('state_sms_change');
                     } else {   // is subscribed for voice messages
-                        return self.states.create('state_voice_menu');
+                        return self.states.create('state_voice_change');
```
Comment (gsvr) [inline src/voice_public.js]: route to exit state in case of problem

File: src/voice_public.js [hunk 2]
```diff
@@ -254,28 +254,35 @@ go.app = function() {
...
             var speech_option = '1';
             return new ChoiceState(name, {
-                question: $('What would you like to do?'),
+                question: $('Please select what you would like to do:'),
                 helper_metadata: go.utils_project.make_voice_helper_data(
                     self.im, name, lang, speech_option),
                 choices: [
                     new Choice('change', $('Change from text to voice')),
-                    new Choice('back', $('Go back to main menu, press 0 then #')),
+                    new Choice('back', $('To go Back to main menu, press 0 then #')),
```
Comment (gsvr) [inline src/voice_public.js]: take this choice out - it's overridden as universal instruction.

File: src/voice_public.js [hunk 3]
```diff
@@ -254,28 +254,35 @@ go.app = function() {
...
                 ],
-                next: 'state_voice_days'
+                next: function(choice) {
+                    if (choice.value === 'change') {
+                        return 'state_voice_days';
+                    } else {
+                        return 'state_sms_change';
+                    }
```
Comment (gsvr) [inline src/voice_public.js]: I think we can replace all of this with just `return 'state_voice_days`

File: src/voice_public.js [hunk 4]
```diff
@@ -336,18 +345,37 @@ go.app = function() {
...
+                question: $('Please select what you would like to do:'),
                 helper_metadata: go.utils_project.make_voice_helper_data(
                     self.im, name, lang, speech_option),
                 choices: [
-                    new Choice('change', $('Change times')),
-                    new Choice('change', $('Change from voice to text')),
-                    new Choice('back', $('Go back to main menu, press 0 then #'))
+                    new Choice('state_voice_days', $('Change times')),
+                    new Choice('state_end_sms_confirm', $('Change mother message from voice to text')),
+                    new Choice('back', $('To go Back to main menu, press 0 then #'))
```
Comment (gsvr) [inline src/voice_public.js]: remove this choice

File: src/voice_public.js [hunk 5]
```diff
@@ -336,18 +345,37 @@ go.app = function() {
...
                 ],
-                next: 'state_baby_save'
+                next: function(choice) {
+                    if (choice.value !== 'back') {
+                        return choice.value;
+                    } else {
+                        return 'state_voice_change';
+                    }
+                }
```
Comment (gsvr) [inline src/voice_public.js:364]: replace with `return choice.value`

**Comment by NicolaasDieBaas** [comment]
@gsvr made the changes you asked.  Should I land this issue?

**Comment by gsvr** [comment]
:+1:

**Pull request merged**